### PR TITLE
[AMDGPU] Check wavefrontsize for GFX11 WMMA builtins

### PR DESCRIPTION
**Pull request by jayfoad**

### CODE REVIEW

File: clang/test/CodeGenOpenCL/builtins-amdgcn-wmma-w64-gfx10-err.cl [hunk 1]
```diff
@@ -21,14 +21,14 @@ void test_amdgcn_wmma_f32_16x16x16_bf16_w64(global v4f* out4f, v16h a16h, v16h b
...
+ *out8s = __builtin_amdgcn_wmma_bf16_16x16x16_bf16_w64(a16s, b16s, c8s, true); // expected-error{{'__builtin_amdgcn_wmma_bf16_16x16x16_bf16_w64' needs target feature gfx11-insts,wavefrontsize64}}
+ *out8h = __builtin_amdgcn_wmma_f16_16x16x16_f16_tied_w64(a16h, b16h, c8h, true); // expected-error{{'__builtin_amdgcn_wmma_f16_16x16x16_f16_tied_w64' needs target feature gfx11-insts,wavefrontsize64}}
+ *out8s = __builtin_amdgcn_wmma_bf16_16x16x16_bf16_tied_w64(a16s, b16s, c8s, true); // expected-error{{'__builtin_amdgcn_wmma_bf16_16x16x16_bf16_tied_w64' needs target feature gfx11-insts,wavefrontsize64}}
+ *out4i = __builtin_amdgcn_wmma_i32_16x16x16_iu8_w64(true, a4i, true, b4i, c4i, false); // expected-error{{'__builtin_amdgcn_wmma_i32_16x16x16_iu8_w64' needs target feature gfx11-insts,wavefrontsize64}}
+ *out4i = __builtin_amdgcn_wmma_i32_16x16x16_iu4_w64(true, a2i, true, b2i, c4i, false); // expected-error{{'__builtin_amdgcn_wmma_i32_16x16x16_iu4_w64' needs target feature gfx11-insts,wavefrontsize64}}
 }
 
-#endif
\ No newline at end of file
+#endif
```
Comment (cdevadas) [inline clang/test/CodeGenOpenCL/builtins-amdgcn-wmma-w64-gfx10-err.cl:34]: Have you added a new line here?

**Comment by piotrAMD** [comment]
Do you think it makes sense to add two gfx11 tests where _w32 variant is now rejected with w64, and _w64 variant rejected with w32? 
Maybe what is being printed in *-gfx10-err.cl test is enough, though.

File: clang/test/CodeGenOpenCL/builtins-amdgcn-wmma-w64-gfx10-err.cl (hunk 1 above)
Comment (jayfoad) [reply clang/test/CodeGenOpenCL/builtins-amdgcn-wmma-w64-gfx10-err.cl:34]: Yes. My editor did that. Previously there was no newline on the end of the `#endif`. Lots of tools flag that as unusual.

**Comment by jayfoad** [comment]
> Do you think it makes sense to add two gfx11 tests where _w32 variant is now rejected with w64, and _w64 variant rejected with w32?

Maybe, but i didn't have the energy to add yet more tests.

> Maybe what is being printed in *-gfx10-err.cl test is enough, though.

Right, that was my thinking.

**Review by piotrAMD — APPROVED** [review]

**Review by cdevadas — APPROVED** [review]

**Review by arsenm — COMMENTED** [review]
Also should get a run line that errors due to wavesize?

**Comment by jayfoad** [comment]
> Also should get a run line that errors due to wavesize?

I think that would mean creating two new *-gfx11-err.cl tests as Piotr was suggesting. Do you really think it's worth it?

**Review by arsenm — APPROVED** [review]

**Pull request merged**